File: Rust/6 Kyu/transform_to_prime.rs
pub fn is_prime(x: u32) -> bool {
    if x == 2 {return true};
    if x < 2 || x % 2 == 0 {return false};
    (3..).step_by(2).take_while(|i| i*i <= x).all(|i| x % i != 0)
}






fn minimum_number(xs: &[u32]) -> u32 {
    let mut soma:u32 = xs.iter().sum();
    let mut cont:u32 = 0;
    
    while !is_prime(soma){
        soma+=1;
        cont+=1;
    }
    cont
}

File: CPP/6 Kyu/transform_to_prime.cpp
#include <numeric>
using namespace std; 
bool isPrime(int x) {
  if (x==2 || x == 3){
      return true;
    }
    if (x<2 || x%2==0){
      return false;
    }
    int lim = round(sqrt(x))+1;
    for (int i = 3; i <= lim;i+=2){
      if(x%i==0){
        return false;
      }
    }
    return true;
}

int minimumNumber (vector <int> numbers )
{
  int total = accumulate(begin(numbers), end(numbers), 0, plus<int>());
  int cont = 0;
  while(!isPrime(total)){
    total++;
    cont++;
    }
  return cont;

}